prelude/src/core/language/Iterator.sk

```diff
@@ -117,6 +117,15 @@ mutable base class .Iterator<+T> {
     }
   }
 
+  overridable mutable fun filterMap<U>(f: T -> ?U): mutable Iterator<U> {
```

Comment:
mapcat - or whatever we call it, flatmap? - is the more general/powerful way of achieving this. Maybe this is more efficient? I imagine it's more ergonomic to use? Just wanted to call this out in case you hadn't considered flatmap.

Reply:
Yes, flatMap would work for this, though we would need Option::iterator(). Like findMap, maybe the compiler could optimize them to the same code, but it would involve flattening a nested loop into an unnested one, so maybe not easy.